tools/pylib/boututils/datafile.py

```diff
@@ -628,6 +629,12 @@ def find_dim(dim):
             # Get new list of variables, and turn into a tuple
             dims = tuple(map(find_dim, dlist))
 
+            if name in self.handle.variables:
+                raise IOError("Variable %s already present",name)
+            tmpname=name+"_bout_temp"
+            i=0
+            while tmpname in self.handle.variables:
+                tmpname=name+"_bout_temp_%d"%i
```

Comment:
Were you meaning to increment `i` here? This looks like suspiciously like an infinite loop